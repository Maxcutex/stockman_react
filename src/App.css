.frame-style {
  box-sizing: "border-box"; padding-bottom: "0px"; border: "none"
}