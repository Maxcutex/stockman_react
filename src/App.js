import React from 'react';
import './assets/css/bootstrap.css';
import './assets/css/style.css';
import './App.css';
import routes from './routes';
import { PersistGate } from 'redux-persist/integration/react'
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import ApplicationTemplate from './Template';

function App() {
    const renderComponents = routes.map(route =>
        <Route
            path={route.path}
            component={route.component}
            exact={route.exact}
            key={route.path}
        />
            );
        return (
            <BrowserRouter>
                <ApplicationTemplate>
                    <Switch>
                        {renderComponents}
                    </Switch>
                </ApplicationTemplate>
            </BrowserRouter>);
}

export default App;
