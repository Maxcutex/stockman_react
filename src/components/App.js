import React, { Component } from "react";
import "../App.css";
import "../assets/css/style.css";
import "../assets/css/bootstrap.css";
import PriceBar from "./common/PriceBar";
import Header from "./common/Header";
import Home from "./Home/Home";

class App extends Component {
  render() {
    return (
      <div>
        <PriceBar />
        <Header />
        <Home />
      </div>
    );
  }
}

export default App;
