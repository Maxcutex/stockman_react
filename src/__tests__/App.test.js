import React from "react";
import { shallow } from "enzyme";
import App from "../components/App";

describe("App", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App />);
  });
  it("should render App correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render Home correctly", () => {
    const home = wrapper.find("Home");
    expect(home.length).toBe(1);
  });
});
