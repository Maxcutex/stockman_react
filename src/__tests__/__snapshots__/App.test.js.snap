// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render App correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <ApplicationTemplate>
        <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/pricelist"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/pricemovement"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/market_price_analysis"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/daily_market_summary"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/asi_movement"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/asi_vs_stocks"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/screened_stocks"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/content-page/:pageType/:id"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/quote-page/:stock_code"
          />
        </Switch>
      </ApplicationTemplate>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/pricelist"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/pricemovement"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/market_price_analysis"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/daily_market_summary"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/asi_movement"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/asi_vs_stocks"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/screened_stocks"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/content-page/:pageType/:id"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/quote-page/:stock_code"
          />
        </Switch>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/pricelist"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/pricemovement"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/market_price_analysis"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/daily_market_summary"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/asi_movement"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/asi_vs_stocks"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/screened_stocks"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/content-page/:pageType/:id"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/quote-page/:stock_code"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "/",
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "/pricelist",
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/pricelist",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "/pricemovement",
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/pricemovement",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "/market_price_analysis",
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/market_price_analysis",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "/daily_market_summary",
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/daily_market_summary",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "/asi_movement",
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/asi_movement",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "/asi_vs_stocks",
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/asi_vs_stocks",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "/screened_stocks",
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/screened_stocks",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "/content-page/:pageType/:id",
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/content-page/:pageType/:id",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "/quote-page/:stock_code",
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/quote-page/:stock_code",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ApplicationTemplate>
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/pricelist"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/pricemovement"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/market_price_analysis"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/daily_market_summary"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/asi_movement"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/asi_vs_stocks"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/screened_stocks"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/content-page/:pageType/:id"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/quote-page/:stock_code"
            />
          </Switch>
        </ApplicationTemplate>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/pricelist"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/pricemovement"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/market_price_analysis"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/daily_market_summary"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/asi_movement"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/asi_vs_stocks"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/screened_stocks"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/content-page/:pageType/:id"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/quote-page/:stock_code"
            />
          </Switch>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/pricelist"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/pricemovement"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/market_price_analysis"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/daily_market_summary"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/asi_movement"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/asi_vs_stocks"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/screened_stocks"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/content-page/:pageType/:id"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/quote-page/:stock_code"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "/",
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "/pricelist",
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/pricelist",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "/pricemovement",
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/pricemovement",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "/market_price_analysis",
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/market_price_analysis",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "/daily_market_summary",
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/daily_market_summary",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "/asi_movement",
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/asi_movement",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "/asi_vs_stocks",
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/asi_vs_stocks",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "/screened_stocks",
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/screened_stocks",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "/content-page/:pageType/:id",
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/content-page/:pageType/:id",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "/quote-page/:stock_code",
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/quote-page/:stock_code",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
